apps/dashboard/app/lib/smart_attributes/attributes/auto_cores.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module SmartAttributes
+    class AttributeFactory
+      # Build this attribute object. No options are used as this Attribute
+      # is meant to be dynamically generated
+      # @param opts [Hash] attribute's options
+      # @return [Attributes::AutoGroups] the attribute object
+      def self.build_auto_cores(opts = {})
+        maximum = 1
+        Configuration.job_clusters.each do |cluster|
+          max_procs = cluster.job_adapter.nodes.max { |a, b| a.procs <=> b.procs }.procs
+          maximum = maximum > max_procs ? maximum : max_procs
```

Comment:
Instead of just setting the min and max blindly, we likely need to add directives so it's per cluster. For example at OSC the max for the `owens` cluster is 28 and for pitzer it's 48. 

Though off the top - I'm not sure if we can achieve this here... Seems like we'd need to apply `data-max-auto-cores: 28` to the cluster option itself, not here.  So I'm not sure off the top how to solve for that.

Reply:
Do we have any existing mechanism for attributes to affect/limit other attributes? If not, maybe setting the max for each cluster here (i.e. `data-max-auto-cores-owens`) and choosing which option to show on the front end could work?

Reply:
> Do we have any existing mechanism for attributes to affect/limit other attributes?

Sure, most things here relate to other things: 

Here's `auto_qos` populating what accounts it's available for and so on.

https://github.com/OSC/ondemand/blob/4b09edb4217269c2a068a3ea3ea036df1a5ce2dd/apps/dashboard/app/lib/account_cache.rb#L110

> If not, maybe setting the max for each cluster here

That's the issue. We can't set the max for each cluster _here_. All the other examples of relationships are between 2 select options. This being a number field - we can't really have the same thing.

I don't think there's anything to do here really. Seems we'll have to add it to add directives to `auto_batch_clusters`.

Reply:
Yea I would say just set the min to 1 and forgo the max. Then we can fixup `auto_batch_clusters` in another PR.